packages/starlight/schemas/sidebar.ts

```diff
@@ -0,0 +1,89 @@
+import type { AstroBuiltinAttributes } from 'astro';
+import type { HTMLAttributes } from 'astro/types';
+import { z } from 'astro/zod';
+import { BadgeConfigSchema } from './badge';
+
+const SidebarBaseSchema = z.object({
+	/** The visible label for this item in the sidebar. */
+	label: z.string(),
+	/** Translations of the `label` for each supported language. */
+	translations: z.record(z.string()).default({}),
+});
+
+const SidebarGroupSchema = SidebarBaseSchema.extend({
+	/** Whether this item should be collapsed by default. */
+	collapsed: z.boolean().default(false),
+});
+
+// HTML attributes that can be added to an anchor element, validated as
+// `Record<string, string | number | boolean | undefined>` but typed as `HTMLAttributes<'a'>`
+// for user convenience.
+const linkHTMLAttributesSchema = z.record(
+	z.union([z.string(), z.number(), z.boolean(), z.undefined()])
+) as z.Schema<Omit<HTMLAttributes<'a'>, keyof AstroBuiltinAttributes | 'children'>>;
```

Comment:
Love this solution! Nice that we can leverage Astro’s built-in type and still keep this pretty lightweight.